src/wwwroot/services/auth.js

```diff
@@ -220,15 +220,51 @@
     }
 
     function getLimitsByAccount() {
+      var accountName = getAccountName();
+
+      if (!accountName) {
+        // limits have no sense in this scenario
+        return $q.when({});
+      }
+
       return $http({
         actionDescription: 'Gathering Free Trial end date',
         method: 'GET',
-        url: RELAY_CONFIG.baseUrl + '/accounts/' + getAccountName()  + '/status/limits'
+        url: RELAY_CONFIG.baseUrl + '/accounts/' + accountName  + '/status/limits'
       })
       .then(function (response) {
-        return response.data;
+        return {
+          monthly: mapLimit(response.data.monthly),
+          daily: mapLimit(response.data.daily),
+          hourly: mapLimit(response.data.hourly),
+          noLimits: !!response.data.noLimits,
+          endDate: mapDate(response.data.endDate)
+        };
       });
-    }   
+    }
+
+    function mapLimit(responseLimit)
+    {
+      if (!responseLimit) {
+        return null;
+      }
+
+      return {
+        limit: responseLimit.limit,
+        remaining: responseLimit.remaining,
+        reset: mapDate(responseLimit.reset)
+      }
+    }
+
+    function mapDate(responseDate)
+    {
+      if (!responseDate) {
+        return null;
+      }
+
+      // TODO: parse date without using moment
```

Comment:
Why is not using moment to parse a date?

Reply:
At the moment we are using moment to parse date and then converting it to a plain old JavaScript date: 
```
string => moment => date
```

I think that it is better to parse directly as date:
```
string => date
```

Or directly deal with moment dates:
```
string => moment
```

Does it make sense?